actions/comment/save.php

```diff
@@ -100,8 +99,13 @@
 	system_message(elgg_echo('generic_comment:posted'));
 }
 
-if ($is_edit_page) {
-	forward($comment->getURL());
+// BC with 1.x: return to activity page if posted from there
+if (!empty($_SERVER['HTTP_REFERER'])) {
+	// don't redirect to URLs from client without verifying within site
+	$site_url = preg_quote(elgg_get_site_url(), '~');
+	if (preg_match("~^{$site_url}activity(/|\\z)~", $_SERVER['HTTP_REFERER'], $m)) {
+		forward("{$m[0]}#elgg-object-{$comment->guid}");
```

Comment:
Might be nice to allow passing in a context to getUrl that gets passed to the hook handlers. In this case we'd do `getUrl('activity')`. We don't want to alter the url handlers here because it would alter URLs in notifications.